src/adapter/src/coord/timestamp_selection.rs

```diff
@@ -111,13 +111,29 @@ impl<S: Append + 'static> Coordinator<S> {
             ));
         };
 
-        Ok(TimestampDetermination {
+        let det = TimestampDetermination {
             timestamp,
             since,
             upper,
             largest_not_in_advance_of_upper,
             oracle_read_ts,
-        })
+        };
+        self.metrics
+            .determine_timestamp
+            .with_label_values(&[
+                match det.respond_immediately() {
+                    true => "true",
+                    false => "false",
+                },
```

Comment:
```suggestion
                format!("{}", det.respond_immediately()),
```

@guswynn this won't allocate?